### ISSUE TITLE
downstream task dyslexia detection using transformer-encoder

### ISSUE THREAD

**Issue by SiQube** [annlaurin/dyslexia_classifier#1]
settings to investigate:
- lstm baseline (with predicted reading time)
- encoder + dense layers trained from scratch
- pre-trained encoder (frozen) + dense layers (fine-tune on downstream task)
- pre-trained encoder (trainable) + dense layers (fine-tune on downstream task) => small learning rate, otherwise you lose the pretraining information (1e-4~1e-5)

encoder(f_1, ..., f_N) = e_1, ..., e_N <= \in [batch, seq, feature]
more settings (everything from above for the encoder):
- average across output embeddings: dense(mean(e_1, ..., e_N)) (from experience <= my bet)
- just first token: dense(e_1)
- just last token: dense(e_N)

add CLS? reading time (?)
- just cls token (if implemented)
nn.Embedding(vocab_size=1, dimension=14) <= learned; learn CLS token.

**Comment by SiQube** [comment]
@annlaurin did you use the downscaled or upscaled dimension as input to classifier (for the transformer) (?)

**Comment by SiQube** [comment]
if you used the "downscaled" you could try using the upscaled as input

**Comment by annlaurin** [comment]
Right, I'll try it out today/tomorrow.

**Comment by annlaurin** [comment]
0.92 +- 0.04

**Comment by SiQube** [comment]
okay -- just add it to the thesis 😓
